module/zfs/txg.c

```diff
@@ -715,7 +715,7 @@ txg_wait_open(dsl_pool_t *dp, uint64_t txg, boolean_t should_quiesce)
 	    txg, tx->tx_quiesce_txg_waiting, tx->tx_sync_txg_waiting);
 	while (tx->tx_open_txg < txg) {
 		cv_broadcast(&tx->tx_quiesce_more_cv);
-		cv_wait(&tx->tx_quiesce_done_cv, &tx->tx_sync_lock);
```

Comment:
Could you add a comment explaining why we are doing cv_wait_sig() but not handling the signal?  Would we want to use _sig in other places as well (e.g. txg_wait_synced())?

Reply:
Sure, I'll add a comment.  We've needed to make this same change, for the same reason, in a few other places as well so I'll audit those and add comments as needed.

Regarding, `txg_wait_synced()` I think we should consider switching it to `cv_wait_io()`.  This was added for Linux and is identical to `cv_wait()`, except it calls `io_schedule()` internally to indicate the blocked process is waiting on IO.  If we do decide that's the right behavior, I'd suggest we do it in a follow up PR.

Reply:
Refreshed.  I annotated some of callers we converted from`cv_wait()` to `cv_wait_sig()` but it felt gratuitous to add the same boiler plate comment to them all.   I think we should consider adding a `porting.txt` document which explains some of these subtle differences between platforms.